users/users_create_test.go

```
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+package users
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"testing"
+
+	"github.com/framey-io/go-tarantool"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zeebo/xxh3"
+
+	"github.com/ice-blockchain/eskimo/users/fixture"
+	messagebrokerfixture "github.com/ice-blockchain/wintr/connectors/message_broker/fixture"
+	. "github.com/ice-blockchain/wintr/testing"
+	"github.com/ice-blockchain/wintr/time"
+)
+
+// nolint:paralleltest,nosnakecase // We cannot use parallel tests in case of empty (=random) referral, cuz of it can fetch referredBy-user from another test
+func TestUserProcessor_CreateUser_Success_FillCountry(t *testing.T) {
+	if testing.Short() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), testDeadline)
+	defer cancel()
+	WHEN("user is created", func() {
+		user := bogusUser(fixture.TestingUserID(0), "").verifyUserCreated(ctx, t, testClientIP)
+		THEN(func() {
+			IT("should be with country and city detected on IP", func() {
+				assert.Equal(t, user.DeviceLocation, DeviceLocation{Country: "US", City: "New York City"})
+			})
+			AND("with hashCode", func() {
+				assert.Equal(t, user.HashCode, xxh3.HashStringSeed(user.ID, uint64(user.CreatedAt.UnixNano())))
+			})
+		})
+	})
+}
+
+// nolint:paralleltest,nosnakecase // We cannot use parallel tests in case of empty (=random) referral, cuz of it can fetch referredBy-user from another test
+func TestUserProcessor_CreateUser_Success_NoReferral(t *testing.T) {
+	if testing.Short() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), testDeadline)
+	defer cancel()
+	SETUP("clear user from another test", func() {
+		// nolint:errcheck // It depends on test execution order, this user is not 100% persisted.
+		_ = usersProcessor.DeleteUser(ctx, fixture.TestingUserID(0))
```

don't ignore any error.

Also, in users package, you can use whatever userID you want, you're not limited to those 10 ones, you can put whatever userID you want